libraries/rush-lib/src/api/SelectorExpressions.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+/**
+ * A "Selector Expression" is a JSON description of a complex selection of
+ * projects, using concepts familiar to users of the Rush CLI. This type represents
+ * the type-safe version of these JSON objects.
+ */
+export type SelectorExpression = ExpressionSelector | ExpressionParameter | ExpressionOperator;
+
+export type ExpressionSelector = string | IExpressionDetailedSelector;
+
+export interface IExpressionDetailedSelector {
+  scope: string;
+  value: string;
+
+  // Reserved for future use
+  filters?: Record<string, string>;
+}
+
+export type ExpressionParameter = Record<`--${string}`, string>;
```

Comment:
Seems like this should be allowed:

<img width="635" alt="Screenshot 2023-09-14 at 1 06 08 AM" src="https://github.com/microsoft/rushstack/assets/107583986/e0b93547-ebae-41e7-a2c6-7d21ea7b819b">